tests/functional/BaseCase.php

```diff
@@ -182,55 +325,44 @@ protected function getStackPrefix(): string
         return $stackPrefix;
     }
 
-    protected function getProjectId(): string
-    {
-        return 'project-' . date('mdHis') . $this->projectSuffix;
-    }
-
-    /**
-     * @return array{GenericBackendCredentials, CreateProjectResponse}
-     */
-    protected function createTestProject(): array
-    {
-        $handler = new CreateProjectHandler($this->clientManager);
-        $command = new CreateprojectCommand();
-
-        $meta = new Any();
-        $meta->pack((new CreateProjectCommand\CreateProjectBigqueryMeta())->setGcsFileBucketName(
-            (string) getenv('BQ_BUCKET_NAME')
-        ));
-        $command->setStackPrefix($this->getStackPrefix());
-        $command->setProjectId($this->getProjectId());
-        $command->setMeta($meta);
-
-        $response = $handler(
-            $this->getCredentials(),
-            $command,
-            [],
-            new RuntimeOptions(),
-        );
-
-        assert($response instanceof CreateProjectResponse);
-
-        return [
-            (new GenericBackendCredentials())
-                ->setPrincipal($response->getProjectUserName())
-                ->setSecret($response->getProjectPassword()),
-            $response,
-        ];
+    private function dropTestBucket(
+        GenericBackendCredentials $projectCredentials,
+        string $bucketId,
+        ?string $branchId
+    ): void {
+        $nameGenerator = new NameGenerator($this->getStackPrefix());
+        $datasetName = $nameGenerator->createObjectNameForBucketInProject($bucketId, $branchId);
+        $bqClient = $this->clientManager->getBigQueryClient($this->testRunId, $projectCredentials);
+        $dataset = $bqClient->dataset($datasetName);
+        try {
+            $dataset->delete(['deleteContents' => true]);
+        } catch (Throwable) {
+            // ignore
+        }
     }
 
     protected function createTestBucket(
-        GenericBackendCredentials $projectCredentials
+        GenericBackendCredentials $projectCredentials,
+        string $projectId,
+        ?string $branchId = null
     ): CreateBucketResponse {
-        $bucket = md5($this->getName()) . 'in.c-Test';
-
+        $bucket = $this->getTestHash() . 'in.c-Test';
+        $this->log->add('Creating bucket:' . $bucket);
+        $this->dropTestBucket(
+            $projectCredentials,
+            $bucket,
+            $branchId
+        );
```

Comment:
vytvoření bucketu funguje stejně jak v connection, prostě per test, vlastně v driveru kopíruju to jak to děláme v connection.